src/App.js

```diff
@@ -0,0 +1,227 @@
+        super();
+        this.state = {
+            todos: [
+                // {id:1 content:"Task1", done:false}
+            ]
+        }
+
+
+    }
+
+    // fetching todos from db.json
+    fetchData = (url) => {
+        fetch(url)
+            .then((response) => {
+                console.log(response);
+                return response
+            })
+            .then((response) => response.json())
+            .then((todos) => this.setState({todos: todos}))
+            .catch((error) => console.error(error));
+    };
+
+    //GET TODOS
+
+    fetchDataHandler = () => {
+        this.fetchData('http://localhost:3000/todos')
+    };
+
+
+    componentWillMount = () => {
+        this.fetchData('http://localhost:3000/todos');
+        console.log('component will mount');
+        console.log(this.state.todos);
+    };
+
+
+    // these method belows are used for CRUD of todos on json-server, not state.
+
+    //POST
+
+    postTodoHandler = (content_input) => {
+        fetch('http://localhost:3000/todos', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                'content': content_input,
+                'done': false
+            })
+        }).then(res => res.json())
+            .then(console.log())
+            .then(() => this.fetchDataHandler());
+    };
+
+    //PUT
+
+    putTodoContentHandler = (event, todo_id, done_stats) => {
+        const todoIndex = this.state.todos.findIndex(todo => {
+            return todo.id === todo_id;
+        });
+        const updatedTodo = {
+            ...this.state.todos[todoIndex]
+        };
+
+        updatedTodo.content = event.target.value;
+
+        const todos = [...this.state.todos];
+        todos[todoIndex] = updatedTodo;
+
+        this.setState({todos: todos});
+
+        let updated_content = event.target.value;
+
+        fetch('http://localhost:3000/todos/' + todo_id, {
+            method: 'PUT',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                'content': updated_content,
+                'done': done_stats
+            })
+        }).then(res => res.json())
+            .then(console.log)
+            .then(() => this.fetchDataHandler());
+    };
+
+
+    putTodoStatusHandler = (todo_id, current_content, done_stats) => {
+        fetch('http://localhost:3000/todos/' + todo_id, {
+            method: 'PUT',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                'content': current_content,
+                'done': !done_stats
+            })
+        }).then(res => res.json())
+            .then(console.log)
+            .then(() => this.fetchDataHandler());
+    };
+
+    deleteTodoHandler = (todo_id) => {
+        fetch('http://localhost:3000/todos/' + todo_id, {
```

Comment:
You can write like this:
```
fetch(`http://localhost:3000/todos/${todoId}`)
```

Reply:
Thank you! I will refactor it.

Reply:
I tried this code but did not work unfortunately. On the browser, the todo content or done status have gone for some reason. I use the original code for now.